pallets/providers/src/types.rs

```diff
@@ -0,0 +1,121 @@
+use codec::{Decode, Encode, FullCodec, HasCompact, MaxEncodedLen};
+use frame_support::traits::fungible::Inspect;
+use frame_support::BoundedVec;
+use frame_support::{
+    pallet_prelude::*,
+    sp_runtime::traits::{AtLeast32BitUnsigned, CheckEqual, MaybeDisplay, SimpleBitOps},
+    traits::fungible,
+};
+use scale_info::prelude::fmt::Debug;
+use scale_info::TypeInfo;
+
+use crate::Config;
+
+/// Structure that has the possible value propositions that a main storage provider can offer (and the runtime is aware of)
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct ValueProposition<T: Config> {
+    pub data_limit: StorageData<T>,
+    pub protocols: BoundedVec<Protocols<T>, MaxProtocols<T>>,
+    // TODO: Add relevant fields here
+}
+
+/// Structure that represents a main storage provider. It holds the amount of data that the MSP is able to store,
+/// the amount of data that it IS storing, its libp2p multiaddress, and its value proposition.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct MainStorageProvider<T: Config> {
+    pub total_data: StorageData<T>,
+    pub data_used: StorageData<T>,
+    pub multiaddress: MultiAddress<T>,
+    pub value_prop: ValueProposition<T>,
+}
+
+/// Structure that represents a backup storage provider. It holds the amount of data that the BSP is able to store,
+/// the amount of data that it is storing, and its libp2p multiaddress.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct BackupStorageProvider<T: Config> {
+    pub total_data: StorageData<T>,
+    pub data_used: StorageData<T>,
+    pub multiaddress: MultiAddress<T>,
+}
+
+// Type aliases:
+
+/// BalanceOf is the balance type of the runtime.
+pub type BalanceOf<T> =
+    <<T as Config>::NativeBalance as Inspect<<T as frame_system::Config>::AccountId>>::Balance;
+
+/// MultiAddress is a byte array that represents the libp2p multiaddress of a storage provider.
+/// Its maximum size is defined in the runtime configuration, as MaxMultiAddressSize.
```

Comment:
The way you've written it here, all this docs comments are going to show in the rust docs just for `MaxMultiAddressSize<T>`. I believe your intention was for them to be docs of `MultiAddress<T>` as well.